src/main/kotlin/nl/joelchrist/shorty/shorties/endpoints/ShortiesEndpoint.kt

```diff
@@ -12,29 +19,55 @@ import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.RequestMethod
 import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.multipart.MultipartFile
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder
 import org.springframework.web.servlet.view.RedirectView
-import javax.validation.Valid
+import java.util.UUID
+import javax.validation.constraints.NotNull
 
 @RestController
 @CrossOrigin
-class ShortiesEndpoint(@Autowired private val shortiesManager: ShortiesManager){
+class ShortiesEndpoint(@Autowired private val shortiesManager: ShortiesManager) {
     companion object {
         private val log = getLogger(ShortiesEndpoint::class)
         private fun getUriFromCurrentRequest() = ServletUriComponentsBuilder.fromCurrentRequest().build().toUri()
     }
 
-    @RequestMapping(value = ["/shorties"],  method = [(RequestMethod.POST)])
-    fun createShorty(@Validated @RequestBody shorty: Shorty) : ResponseEntity<Shorty> =
+    @RequestMapping(value = ["/shorties"], method = [(RequestMethod.POST)])
+    fun createShorty(@Validated @RequestBody shorty: Shorty): ResponseEntity<Shorty> =
             ResponseEntity
                     .created(getUriFromCurrentRequest())
-                    .body(shorty.takeIf { it.identifier != null }?.let(shortiesManager::create) ?: shorty.url.let(shortiesManager::create))
-                    .also { log.info("Creating shorty for url: ${shorty.url}") }
+                    .body(shorty.takeIf { it.identifier != null }?.let(shortiesManager::create) ?: shorty.url.let(shortiesManager::create).let(shortiesManager::create))
+                    .also { log.info("Creating shorty for url ${shorty.url}") }
 
-    @RequestMapping(value = ["/{identifier}"],  method = [(RequestMethod.GET)])
-    fun useShorty(@PathVariable(value = "identifier") identifier: String) : RedirectView =
+    @RequestMapping(value = ["/files"], method = [(RequestMethod.POST)])
+    fun createFile(@Validated @NotNull @NotBlank file: MultipartFile): ResponseEntity<ShortyMetadata> =
+            ResponseEntity
+                    .created(getUriFromCurrentRequest())
+                    .body(UUID.randomUUID().let { ShortyMetadata(shortiesManager.create(it, file), it.toString()) })
+                    .also { log.info("Creating file ${file.originalFilename}") }
+
+    @RequestMapping(value = ["/{identifier}"], method = [(RequestMethod.GET)])
+    fun useShorty(@PathVariable(value = "identifier") identifier: String): RedirectView =
             RedirectView()
                     .also { it.url = shortiesManager.find(identifier).url }
                     .also { log.info("Using shorty with identifier $identifier") }
 
+    @RequestMapping(value = ["files/{identifier}"], method = [(RequestMethod.GET)])
```

Comment:
Kan dit echt niet in hetzelfde endpoint als `/{identifier}`? Dat vind ik wel het mooist.

Reply:
Vind ik ook, maar kan niet want ik moet custom headers en content-type zetten en Spring vind mixed content-types niet leuk.